src/main/java/org/prgrms/kdtspringdemo/customer/service/CustomerService.java

```diff
@@ -0,0 +1,66 @@
+package org.prgrms.kdtspringdemo.customer.service;
+
+import org.prgrms.kdtspringdemo.customer.exception.CustomerIdNotFoundException;
+import org.prgrms.kdtspringdemo.customer.exception.CustomerNicknameNotFoundException;
+import org.prgrms.kdtspringdemo.customer.model.dto.CustomerResponse;
+import org.prgrms.kdtspringdemo.customer.model.entity.Customer;
+import org.prgrms.kdtspringdemo.customer.repository.CustomerRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static org.prgrms.kdtspringdemo.customer.exception.CustomerExceptionMessage.*;
+
+@Service
+public class CustomerService {
+    private final CustomerRepository customerRepository;
+
+    public CustomerService(CustomerRepository customerRepository) {
+        this.customerRepository = customerRepository;
+    }
+
+    public CustomerResponse create(String nickname) {
+        Customer savedCustomer = customerRepository.save(new Customer(nickname));
+
+        return CustomerResponse.toDto(savedCustomer.getCustomerId(), savedCustomer.getNickname());
+    }
+
+    public CustomerResponse findById(UUID customerID) {
+        Customer customer = validateExist(customerRepository.findById(customerID));
+
+        return CustomerResponse.toDto(customer.getCustomerId(), customer.getNickname());
+    }
+
+    private static Customer validateExist(Optional<Customer> foundCustomer) {
+        return foundCustomer.orElseThrow(() -> new CustomerIdNotFoundException(CUSTOMER_ID_LOOKUP_FAILED));
+    }
+
+    public CustomerResponse findByNickname(String nickname) {
+        Optional<Customer> foundCustomer = customerRepository.findByNickname(nickname);
+        Customer customer = foundCustomer.orElseThrow(() -> new CustomerNicknameNotFoundException(CUSTOMER_NICKNAME_LOOKUP_FAILED));
+
+        return CustomerResponse.toDto(customer.getCustomerId(), customer.getNickname());
+    }
+
+    public List<CustomerResponse> findAll() {
+        return customerRepository.findAll()
+                .stream()
+                .map(v -> new CustomerResponse(v.getCustomerId(), v.getNickname()))
+                .collect(Collectors.toUnmodifiableList());
```

Comment:
개인 취향인 것 같지만 stream()의 연산 시작부터 라인 바이 라인으로 적용하는건 어떻게 생각하시나요?
```suggestion
        return customerRepository.findAll().stream()
                .map(v -> new CustomerResponse(v.getCustomerId(), v.getNickname()))
                .collect(Collectors.toUnmodifiableList());
```

Reply:
[6007d96](https://github.com/prgrms-be-devcourse/springboot-basic/pull/838/commits/6007d96a3b30f10e586cd0da5bf40523ee744a47) 피드백 적용
- 감사합니다! 개행처리를 지움으로써 코드 라인을 줄일 수 있었습니다!!🪄🪄